### ISSUE TITLE
`reticulate` issue with `joblib` (and consequently `sklearn`) on Windows

### ISSUE THREAD

**Issue by Unco3892** [rstudio/reticulate#1346]
Hi, thanks for your great work. Calling python directly with `joblib` (conda environment) works fine. But when I try it in an interactive session with `reticulate`, be it from Rstudio or command line, it crashes. 

In an interactive session (Rstudio) I get the following error:
```
> reticulate::repl_python()
Python 3.8.16 (C:/Users/iazizi/.conda/envs/ML_2023/python.exe)
Reticulate 1.26 REPL -- A Python interpreter in R.
Enter 'exit' or 'quit' to exit the REPL and return to R.
>>> from joblib import Parallel, delayed
>>> from math import modf
>>> print(Parallel(n_jobs=2)(delayed(modf)(i/2.) for i in range(10)))
exception calling callback for <Future at 0x1e21d9b2430 state=finished raised TerminatedWorkerError>
Traceback (most recent call last):
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\externals\loky\_base.py", line 26, in _invoke_callbacks
    callback(self)
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\parallel.py", line 385, in __call__
    self.parallel.dispatch_next()
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\parallel.py", line 834, in dispatch_next
    if not self.dispatch_one_batch(self._original_iterator):
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\parallel.py", line 901, in dispatch_one_batch
    self._dispatch(tasks)
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\parallel.py", line 819, in _dispatch
    job = self._backend.apply_async(batch, callback=cb)
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\_parallel_backends.py", line 556, in apply_async
    future = self._workers.submit(SafeFunction(func))
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\externals\loky\reusable_executor.py", line 176, in submit
    return super().submit(fn, *args, **kwargs)
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\externals\loky\process_executor.py", line 1129, in submit
    raise self._flags.broken
joblib.externals.loky.process_executor.TerminatedWorkerError: A worker process managed by the executor was unexpectedly terminated. This could be caused by a segmentation fault while calling the function or by an excessive memory usage causing the Operating System to kill the worker.

C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\externals\loky\backend\utils.py:62: UserWarning: Failed to kill subprocesses on this platform. Please installpsutil: https://github.com/giampaolo/psutil
Details:
Traceback (most recent call last):
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\externals\loky\backend\utils.py", line 57, in _kill_process_tree_without_psutil
    _windows_taskkill_process_tree(process.pid)
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\site-packages\joblib\externals\loky\backend\utils.py", line 80, in _windows_taskkill_process_tree
    subprocess.check_output(
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\subprocess.py", line 493, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\subprocess.py", line 808, in __init__
    errread, errwrite) = self._get_handles(stdin, stdout, stderr)
  File "C:\Users\iazizi\CONDA~1\envs\ML_2023\lib\subprocess.py", line 1146, in _get_handles
    p2cread = _winapi.GetStdHandle(_winapi.STD_INPUT_HANDLE)
OSError: [WinError 6] The handle is invalid

  warnings.warn(
TerminatedWorkerError: A worker process managed by the executor was unexpectedly terminated. This could be caused by a segmentation fault while calling the function or by an excessive memory usage causing the Operating System to kill the worker.

>>> exit
```
From the command line, `source_python()` does not even produce an error and R session crashes and restarts immediately. I was using this with `sklearn` for when you want to set parallel jobs (e.g. from `sklearn.model_selection.GridSearchCV`). Here's my session info and python configuration.

```
> sessionInfo()
R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19043)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] reticulate_1.26

loaded via a namespace (and not attached):
 [1] compiler_4.2.1  here_1.0.1      Matrix_1.5-3    rprojroot_2.0.3 tools_4.2.1     rstudioapi_0.14
 [7] rappdirs_0.3.3  Rcpp_1.0.10     grid_4.2.1      jsonlite_1.8.4  renv_0.16.0     png_0.1-8      
[13] lattice_0.20-45
> py_config()
python:         C:/Users/iazizi/.conda/envs/ML_2023/python.exe
libpython:      C:/Users/iazizi/.conda/envs/ML_2023/python38.dll
pythonhome:     C:/Users/iazizi/.conda/envs/ML_2023
version:        3.8.16 | packaged by conda-forge | (default, Feb  1 2023, 15:53:35) [MSC v.1929 64 bit (AMD64)]
Architecture:   64bit
numpy:          C:/Users/iazizi/AppData/Roaming/Python/Python38/site-packages/numpy
numpy_version:  1.23.5

NOTE: Python version was forced by use_python function
```
Thank you in advance for your help

**Comment by t-kalinowski** [comment]
Thank you for the report! I can reproduce this on Windows w/ reticulate, but not on Linux, and not on Windows with without reticulate. Also, this reproduces outside of conda, so it's unlikely to be a binary incompatibility issue. I think this is a genuine reticulate bug.

I looked into the joblib sources briefly, but it will require a deeper investigation to figure out what's going on. Given that this only affects one platform (Windows), and there is an accessible workaround (using WSL), I don't think we'll have time to track this one down in the near future. Contributions are welcome!

**Comment by t-kalinowski** [comment]
Not sure if this was fixed by #1458, or by #1461 (the error message hints at both being contributing factors), but this is fixed now.

In `repl_python()`, this now gives the expected output:
```python
from joblib import Parallel, delayed
from math import modf

print(Parallel(n_jobs=2)(delayed(modf)(i/2.) for i in range(10)))
```
```
[(0.0, 0.0), (0.5, 0.0), (0.0, 1.0), (0.5, 1.0), (0.0, 2.0), (0.5, 2.0), (0.0, 3.0), (0.5, 3.0), (0.0, 4.0), (0.5, 4.0)]
```
